documentation/modules/managing/con-consumer-config-properties.adoc

```diff
@@ -193,17 +231,18 @@ isolation.level=read_committed <1>
 
 == Recovering from failure to avoid data loss
 
-Use the `session.timeout.ms` and `heartbeat.interval.ms` properties to configure the time taken to check and recover from consumer failure within a consumer group.
-
-The `session.timeout.ms` property specifies the maximum amount of time in milliseconds a consumer within a consumer group can be out of contact with a broker before being considered inactive and a _rebalancing_ is triggered between the active consumers in the group.
-When the group rebalances, the partitions are reassigned to the members of the group.
+Kafka provides a rebalance protocol which helps with failure detection and recovery within a consumer group.  
+Two useful properties that aid in this recovery process are `session.timeout.ms` and `heartbeat.interval.ms`.
```

Comment:
In most cases users should not have to tweak these settings. They can be used for relatively advanced use cases if there's a good justification, in my experience tweaking these early is often a red flag and instead tuning `max.poll.records` is preferred.

Reply:
I'v changed the priority of this section to put the emphasis on `max.poll.records`